samples: minimal: refactor meta data

hakehuang opened the pull request:
remove platform_allow, extend integration_platform, and try to use feature tag to extend support for platforms

hakehuang committed:
samples: minimal: refactor meta data

remove platform_allow, extend integration_platform

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang commented:
with this change there is a minor issue for qemu_x86_64_atom.

![image](https://github.com/user-attachments/assets/5f34a147-e2ce-4635-93fe-7967279202b6)

nashif commented:
> with this change there is a minor issue for qemu_x86_64_atom.

try `./scripts/twister -T samples/basic/minimal/ --all` and report back please. 

> remove platform_allow, extend integration_platform

why is this needed? what do we get out of this?

nashif commented on samples/basic/minimal/sample.yaml:
```diff
@@ -9,81 +9,86 @@ common:
-    extra_args: CONF_FILE='common.conf;mt.conf;arm.conf'
+  sample.minimal.mt:
+    arch_allow:
+      - arm
```
why `arch_allow` is good but not `plaform_allow`?

butok replied:
> why `arch_allow` is good but not `platform_allow`?

This is better than adding a list of all arm and x86 boards.

hakehuang replied:
I find arc and riscv has its own minimal test config already. so I assumption is that the minimal setting config is arch depended.
```
  sample.minimal.arc.runtime:
    extra_args:
      - CONF_FILE='common-runtime.conf;no-timers.conf'
    arch_allow: arc
    tags:
      - kernel
    integration_platforms:
      - qemu_arc/qemu_arc_em
  sample.minimal.riscv.runtime:
    extra_args:
      - CONF_FILE='common-runtime.conf;no-timers.conf;no-mt.conf;riscv.conf'
    platform_allow: qemu_riscv32
    tags:
      - kernel
    integration_platforms:
      - qemu_riscv32
```

hakehuang committed:
soc: NXP: add default CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC

if CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC is not config a default value,
it will just defined, and will have below error

zephyr/include/zephyr/sys_clock.h:158:45:
  error: operator '==' has no left operand
  158 |         (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC == 0)

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang committed:
lib: os : printk : fix a config depency issue

fix a printk dependency issue

printk.c:33:26: error: 'lock' defined but not used

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang commented:
@nashif I add fixes for integration platforms, and some of NXP platforms. please review. Thanks

hakehuang committed:
doc: tag: add minimal tags for filter minimal

add minimal as tag for platforms

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

kartben reviewed (changes requested):
Introducing a supported tag for indicating whether a board can run a particular sample doesn't sound like a solution that scales when there are 500 samples potentially calling for the same "supported" tags?

hakehuang commented:
> Introducing a supported tag for indicating whether a board can run a particular sample doesn't sound like a solution that scales when there are 500 samples potentially calling for the same "supported" tags?

This is only for minimal, as discussion in the https://github.com/zephyrproject-rtos/zephyr/issues/85359, we can not remove the platform_allow, and we do not want this to be kept adding. Then tag is the only way to move platforms in, for now only, integration_platform and NXP platforms are enabled.

hakehuang committed:
soc: nxp: mcxw: fix including issue

for soc only fls_common.h need to be include, and port
need depends on usage

fixes: #85923

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang committed:
driver: clock: fix clock_control_mcux_syscon define error

the define in wrong sequence

fixes: 85924

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang committed:
soc: mcxc: include fixes

include port fsl_port.h when needed

fixes: #85923

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

nashif commented:
> This is only for minimal, as discussion in the #85359, we can not remove the platform_allow, and we do not want this to be kept adding. Then tag is the only way to move platforms in, for now only, integration_platform and NXP platforms are enabled.

The discussion in #85359 is still ongoing, no conclusions or recommendations were made, lets not get ahead of ourselves with solutions like this.

hakehuang commented:
> The discussion in #85359 is still ongoing, no conclusions or recommendations were made, lets not get ahead of ourselves with solutions like this.

@nashif , I see, I move this PR to draft. and the issues I found with this PR are real issue, if you look at my fixes, many coding issues and dependency issue found.

hakehuang committed:
samples: minimal: disable SHELL

add CONFIG_SHELL=n as some defconf enable it

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

hakehuang commented:
> This PR contains several useful, independent improvements/fixes. It would probably be better to split it into several PRs to help them get approved by the community and avoid having the entire PR blocked by a single change.

ok, let me create new pr for fixes only
